main.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+function isInteger(num) {
+  return (num ^ 0) === num;
+}
+
+function dateVlaidation(month, year){
+  if(isInteger(year) && isInteger(month)) {
+    if(month >= 1 && month <= 12) {

```

Comment:
do not add unneeded nesting
if (
    isInteger(year)
    && isInteger(month) 
    && month >= 1 
    && month <= 12 
) {